homeassistant/components/deconz/light.py

```diff
@@ -208,6 +223,13 @@ def __init__(self, device: _LightDeviceT, hub: DeconzHub) -> None:
         if device.effect is not None:
             self._attr_supported_features |= LightEntityFeature.EFFECT
             self._attr_effect_list = [EFFECT_COLORLOOP]
+
+            # For lights that report supported effects.
+            if isinstance(device, Light) and device.supported_effects is not None:
+                self._attr_effect_list = [
+                    EFFECT_TO_DECONZ[el.value] for el in device.supported_effects
+                ]
```

Comment:
What if EFFECT_TO_DECONZ doesn't contain a newer value supported by deconz?
Also the LightEffect is a StrEnum so shouldn't need to use el.value here
```suggestion
                    EFFECT_TO_DECONZ[el] for el in device.supported_effects
                    if el in EFFECT_TO_DECONZ
                ]
```